src/Cohesity.Powershell/Scripts/Routes/Get-CohesityRoutes.ps1

```diff
@@ -0,0 +1,78 @@
+function Get-CohesityRoutes {
+    <#
+        .SYNOPSIS
+        Get the routes.
+        .DESCRIPTION
+        The Get-CohesityRoutes function is used to get routes.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Get-CohesityRoutes
+        .EXAMPLE
+        Get-CohesityRoutes -FilterName INTERFACE-GROUP-NAME -FilterValue "intf_group1"
+        .EXAMPLE
+        Get-CohesityRoutes -FilterName DESTINATION-NETWORK -FilterValue "1.2.4.14/32"
+    #>
+    [CmdletBinding(DefaultParameterSetName = 'Default')]
+    Param(
+        [Parameter(Mandatory = $true, ParameterSetName = 'Filter')]
+        [ValidateSet("DESTINATION-NETWORK", "NEXT-HOP", "INTERFACE-GROUP-NAME")]
+        [ValidateNotNullOrEmpty()]
+        $FilterName,
+        [Parameter(Mandatory = $true, ParameterSetName = 'Filter')]
+        [ValidateNotNullOrEmpty()]
+        $FilterValue
+    )
+
+    Begin {
+        if (-not (Test-Path -Path "$HOME/.cohesity")) {
+            throw "Failed to authenticate. Please connect to the Cohesity Cluster using 'Connect-CohesityCluster'"
+        }
+        $cohesitySession = Get-Content -Path $HOME/.cohesity | ConvertFrom-Json
+        $cohesityServer = $cohesitySession.ClusterUri
+        $cohesityToken = $cohesitySession.Accesstoken.Accesstoken
+    }
+
+    Process {
+        $cohesityUrl = $cohesityServer + '/irisservices/api/v1/public/routes'
+        $cohesityHeaders = @{'Authorization' = 'Bearer ' + $cohesityToken }
+        $resp = Invoke-RestApi -Method Get -Uri $cohesityUrl -Headers $cohesityHeaders
+
+        if ($FilterName -and $FilterValue) {
```

Comment:
Both are mandatory field, and checking for non-null value right?
Then why we need to check for 'If' condition?

Reply:
Filter is mutually exclusive with respect to get all operation. The examples describe the same.

Reply:
I agree with Kavishree here.. why is the condition if both are manadatory.

Reply:
There are 2 ways that the cmdlet can be executed.
1. Get-CohesityRoutes (which will get all the routes objects)
2. Get-CohesityRoutes -FilterName INTERFACE-GROUP-NAME -FilterValue "intf_group1" (which will get all the filtered objects)

We have the if condition to check for the #2 case (ParameterSetName = 'Filter') and else #1 (ParameterSetName = 'Default')

Let me know if we can write it in a better way.

Reply:
since either parameters are mandatory anyways .. you could check only 1 parameter ? Feel free to retain.